### ISSUE TITLE
lef write -hide doesn't always create obstructions

### ISSUE THREAD

**Issue by antonblanchard** [RTimothyEdwards/magic#204]
In https://github.com/The-OpenROAD-Project/OpenLane/issues/1522 it was noticed that the LEFs created by magic sometimes have an obstruction on met4, and sometimes not. The test case has two examples. In the first one we see no met4 obstructions at all:

![met4_obs_1](https://user-images.githubusercontent.com/2029971/204687189-eff852a2-a282-4999-8cc7-b41d93c5f723.png)

The second example has some of the met4 power strips removed, and in that case we partially create a met4 obstruction:

![met4-obs-2](https://user-images.githubusercontent.com/2029971/204687311-57c524b2-8bfc-464c-b4b3-aaf7aa80b4b3.png)

I took a look at magic but wasn't able to work out what is going on.

Test case: [lef-issue.tar.gz](https://github.com/RTimothyEdwards/magic/files/10118611/lef-issue.tar.gz)

**Comment by RTimothyEdwards** [comment]
What's going on is that magic is looking at all geometry which would create obstructions---which is anything on a metal layer that is not a pin---and if the "-hide" option is chosen, then it will generate an obstruction layer matching the bounding box of the obstructions.  In case 1, you have only pins on metal4, so there is no additional material and so there is no obstruction created.  In case 2, the wires that are not marked as pins are obstructions to be avoided, so the "-hide" option creates an obstruction encompassing all of them, with cut-outs for the pins.

In other words, it's behaving as intended.  You could work around this by adding BLOCKAGE layers to the DEF file, or (equivalently) adding obstruction layers to the layout.  I could also do something like implement a check for some property, say "OBS_BBOX", that would be used to enforce a specific area to be marked as an obstruction in the output LEF.
